src/kernel.ts

```diff
@@ -884,12 +905,15 @@ class Kernel implements IKernel {
       return;
     }
     var content = msg.content as ICommOpen;
-    if (!content.target_module) {
-      this.commOpened.emit(msg);
+    this.commOpened.emit(msg);
+    var targetName = content.target_name;
+    var moduleName = content.target_module;
+    if (this._targetRegistry[targetName]) {
+      this._targetRegistry[targetName](msg);
+    }
```

Comment:
Sylvain and I are talking, and he's convincing me that maybe we don't want *any* registry, and just have `(module, target)` based loaders...